usd/src/usd_parser/USDWorld.cc

```diff
@@ -50,10 +62,65 @@ namespace usd
     }
     _world->worldName = reference->GetDefaultPrim().GetName().GetText();
 
+    std::string linkName;
+
     auto range = pxr::UsdPrimRange::Stage(reference);
     for (auto const &prim : range)
     {
+      // Skip materials, the data is already available in the USDData class
+      if (prim.IsA<pxr::UsdShadeMaterial>() || prim.IsA<pxr::UsdShadeShader>())
+      {
+        continue;
+      }
+
       std::string primName = prim.GetName();
+      std::string primPath = pxr::TfStringify(prim.GetPath());
+      std::string primType = pxr::TfStringify(prim.GetPath());
+
+      std::vector<std::string> primPathTokens =
+        ignition::common::split(primPath, "/");
+
+      // In general USD models used in Issac Sim define the model path
+      // under a root path for example:
+      //  -> /robot_name/robot_name_link0
+      // But sometimes for enviroments it uses just a simple path:
+      //  -> /ground_plan
+      //  -> /wall_0
+      // the shortName variable defines if this is the first case when it's
+      // False or when it's true then it's the second case.
+      if (primPathTokens.size() >= 2)
```

Comment:
The usd input may not come from isaac sim and so it may not be following isaac sim convention. We should probably note somewhere that the USD to SDF converter only works for isaac sim USDs.